openmetadata-service/src/main/java/org/openmetadata/service/migration/utils/v172/MigrationUtil.java

```diff
@@ -0,0 +1,115 @@
+package org.openmetadata.service.migration.utils.v172;
+
+import java.util.List;
+import lombok.SneakyThrows;
+import lombok.extern.slf4j.Slf4j;
+import org.openmetadata.common.utils.CommonUtil;
+import org.openmetadata.service.search.SearchClient;
+import org.openmetadata.service.search.SearchRepository;
+
+@Slf4j
+public class MigrationUtil {
+  private MigrationUtil() {}
+
+  private static final String DATA_INSIGHTS_PREFIX = "di-data-assets";
+  private static final String INDEX_TEMPLATE_NAME = "di-data-assets";
+  private static final String ILM_POLICY_NAME = "di-data-assets-lifecycle";
+  private static final List<String> COMPONENT_TEMPLATES_NAMES =
+      List.of("di-data-assets-mapping", "di-data-assets-settings");
+  private static String clusterAlias;
+
+  public static void removeOldDataInsightsObjects() {
+    // From 1.6.6 we implemented the support for CLUSTER_ALIAS for Data Insights.
+    // The old objects without CLUSTER_ALIAS were not cleaned then.
+    // From 1.7.1 we removed the ILM policy.
+    LOG.info("Starting cleanup of old Data Insights objects");
+    SearchRepository searchRepository = org.openmetadata.service.Entity.getSearchRepository();
+    SearchClient searchClient = searchRepository.getSearchClient();
+    clusterAlias = searchRepository.getClusterAlias();
+
+    if (clusterAlias != null && !clusterAlias.isEmpty()) {
+      // Delete data insights objects without cluster alias
+      try {
+        deleteDataInsightsDataStreams(searchClient);
+        deleteIndexTemplate(searchClient);
+        deleteComponentTemplate(searchClient);
+        deleteIlmPolicy(searchClient, null);
+        LOG.info("Successfully completed Data Insights objects cleanup");
+      } catch (Exception e) {
+        LOG.error("Error deleting Data Insights objects", e);
+        throw e;
+      }
+    } else {
+      LOG.info("No CLUSTER_ALIAS found, skipping cleanup");
+    }
+
+    deleteIlmPolicy(searchClient, clusterAlias);
+    LOG.info("Successfully completed Data Insights ILM cleanup");
+  }
+
+  private static String getClusteredPrefix(String clusterAlias, String prefix) {
+    if (CommonUtil.nullOrEmpty(clusterAlias)) {
+      return prefix;
+    } else {
+      return String.format("%s-%s", clusterAlias, prefix);
+    }
+  }
+
+  @SneakyThrows
+  private static void deleteDataInsightsDataStreams(SearchClient searchClient) {
+    try {
+      // Get and delete data streams that don't have cluster alias
+      List<String> dataStreams =
+          searchClient.getDataStreams(String.format("%s-*", DATA_INSIGHTS_PREFIX));
+      for (String name : dataStreams) {
```

Comment:
lets review the behavior of any of these ops to ensure they don't fail in the second run (e.g., imagine force migrations, or if a cluster is "new" and does not have old ILM etc.). We should manage these cases gracefully and log accordingly